### PR TITLE
steamPackages.steam-udev: init at 1.0.0.51

### LINKED ISSUE #47314: Create UDEV rules for Steam Controller
For the Steam Controller to work correctly, you need to add several udev rules. Those rules are included in the archive fetched to create the steam package : https://github.com/NixOS/nixpkgs/blob/a729229d19445750a313e82a74f79845041786b4/pkgs/games/steam/steam.nix#L11

Would it be possible to create an udev package with those rules ? I can't think of the best way to achieve this.

Help would be much appreciated :)

### PR DESCRIPTION
**Pull request by nyanloutre**
###### Motivation for this change
Fixes #47314

I would like to have some input on the way I did this.
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by nyanloutre** [comment]
Next step would be to create a module to activate those rules easily

File: pkgs/games/steam/steam-udev.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+stdenv.mkDerivation rec {
+  name= "steam-udev-${version}";
+  version = lib.getVersion steamPackages.steam.name;
+
+  src = steamPackages.steam.src;
+ 
+  installPhase = ''
+    install -d $out/lib/udev/rules.d
+    install -m644 lib/udev/rules.d/*.rules $out/lib/udev/rules.d
```
Comment (Mic92) [inline pkgs/games/steam/steam-udev.nix]: Why not exposing the rules in the original `steamPackages.steam`?

Comment (Mic92) [reply pkgs/games/steam/steam-udev.nix]: It could be added as udev rule package in a nixos module as well.

Comment (nyanloutre) [reply pkgs/games/steam/steam-udev.nix]: If you say it's a good way to do it I will do a new pull request

**Comment by Mic92** [comment]
I prefer the other one.

**Pull request closed without merging**